catalog/controller/payment/razorpay.php

```diff
@@ -105,6 +105,18 @@ public function index(): string {
                 $data['is_recurring'] = "false";
                 // Orders API with payment autocapture
                 $order_data = $this->get_order_creation_data($this->session->data['order_id']);
+
+                if ($order_info['order_status_id'] and
+                    isset($this->session->data["razorpay_order_id_" . $this->session->data['order_id']]) === true)
+                {
+                    $rzpOrder = $this->api->order->fetch($this->session->data["razorpay_order_id_" . $this->session->data['order_id']]);
+
+                    if ($rzpOrder['status'] === 'paid')
+                    {

```

Comment:
@abdulwahidsharief , can't we clear the carts items here with session?

Reply:
@ramth05 on redirect to success page opencart clears all cart items and session data